Index: g_editor.c
===================================================================
RCS file: /cvsroot/pure-data/pd/src/g_editor.c,v
retrieving revision 1.20.6.3
diff -u -w -r1.20.6.3 g_editor.c
--- g_editor.c	9 Nov 2007 06:28:44 -0000	1.20.6.3
+++ g_editor.c	26 Nov 2007 22:51:05 -0000
@@ -1420,8 +1420,6 @@
                 gobj_activate(x->gl_editor->e_selection->sel_what,
                     x, 1);
     }
-    if (x->gl_editor->e_onmotion != MA_NONE)
-        sys_vgui("pdtk_canvas_getscroll .x%lx.c\n", x);
 
     x->gl_editor->e_onmotion = MA_NONE;
 }
@@ -1446,6 +1444,7 @@
     }
     if (resortin) canvas_resortinlets(x);
     if (resortout) canvas_resortoutlets(x);
+    sys_vgui("pdtk_canvas_getscroll .x%lx.c\n", x);
     canvas_dirty(x, 1);
 }
 
