Index: g_canvas.c
===================================================================
--- g_canvas.c	(revision 9721)
+++ g_canvas.c	(working copy)
@@ -598,9 +598,15 @@
         strcat(namebuf, ")");
     }
     else namebuf[0] = 0;
+#ifdef __APPLE__
+    sys_vgui("wm attributes .x%lx -modified %d -titlepath {%s/%s}\n",
+        x, x->gl_dirty, canvas_getdir(x)->s_name, x->gl_name->s_name);
+    sys_vgui("wm title .x%lx {%s%s}\n", x, x->gl_name->s_name, namebuf);
+#else
     sys_vgui("wm title .x%lx {%s%c%s - %s}\n", 
         x, x->gl_name->s_name, (x->gl_dirty? '*' : ' '), namebuf,
             canvas_getdir(x)->s_name);
+#endif
 }
 
 void canvas_dirty(t_canvas *x, t_int n)
