Index: g_array.c
===================================================================
--- g_array.c	(revision 9700)
+++ g_array.c	(working copy)
@@ -1207,7 +1207,7 @@
 
 static void garray_sinesum(t_garray *x, t_symbol *s, int argc, t_atom *argv)
 {    
-    t_float *svec = (t_float *)t_getbytes(sizeof(t_float) * argc);
+    t_float *svec;
     int npoints, i;
     if (argc < 2)
     {
@@ -1230,7 +1230,7 @@
 
 static void garray_cosinesum(t_garray *x, t_symbol *s, int argc, t_atom *argv)
 {
-    t_float *svec = (t_float *)t_getbytes(sizeof(t_float) * argc);
+    t_float *svec;
     int npoints, i;
     if (argc < 2)
     {
