diff --git a/packages/Makefile b/packages/Makefile
index a360dba7..b5b949b5 100644
--- a/packages/Makefile
+++ b/packages/Makefile
@@ -377,7 +377,7 @@ doc_format:
 # from the latter by substituting PD_BUILD_VERSION.
 
 git_version: $(pd_src)/src/s_stuff.h.in
-	@cd $(pd_src)/src/ && \
+	@test -d $(cvs_root_dir)/.git && cd $(pd_src)/src/ && \
 	sed 's|^\(#define PD_BUILD_VERSION "\).*"|\1$(PD_BUILD_VERSION)"|' s_stuff.h.in > s_stuff.h.new && \
 	(test -f s_stuff.h && diff -q s_stuff.h s_stuff.h.new > /dev/null && rm -f s_stuff.h.new && echo s_stuff.h unchanged || (rm -f s_stuff.h && mv s_stuff.h.new s_stuff.h))
 
@@ -422,7 +422,7 @@ distclean: cruft_clean
 	$(MAKE) -C $(abstractions_src) distclean
 	$(MAKE) -C $(extensions_src) distclean
 	$(MAKE) -C $(externals_src) distclean
-	-$(MAKE) -C $(pd_src) distclean
+	-$(MAKE) -C $(pd_src) clean
 
 test_locations:
 	@echo "PD_VERSION: $(PD_VERSION)"
